### DISCUSSION TITLE
Async treasure map generation?

### DISCUSSION THREAD

**Discussion by BigScary** [PaperMC/Paper#10736] [Ideas]
### Is your feature request related to a problem?

I recently tuned my Paper server very carefully using an up to date (1.20) guide. I was able to get great performance overall, but I and my players were super disappointed that one of the biggest recommendations was disabling treasure maps, a big blow to game experience.

### Describe the solution you'd like.

Maybe it's just a pie in the sky and not really possible. But I would love to hear news that actually there's some other less impactful alternative to avoiding lag in my case, or that moving the map generation off the main server thread is possible. <3

### Describe alternatives you've considered.

Using the Plan plugin to measure performance over the last 30 days, I have determined that I must choose one of the three to avoid significant lag:

1. Greatly reduce view distance (currently at 8, on a server all about flying around with elytras).
2. Cap players at 20 (we would otherwise hit 30+ several times per week).
3. Disable treasure maps.

**Comment by electronicboy** [comment]
The problem is that the generation happens when you open the chest, and so either you have to block people from opening the chest too early, or give them a dud map in hopes that you can safely update the thing properly later. Mojang is apparently drastically improving this again, so this should be less of a concern.

**Reply by BigScary** [reply]
Sounds promising, thanks much for the update. I'll stay tuned. <3
